source/connectors/kafka-sink-postprocessors.txt

```diff
@@ -9,10 +9,821 @@ Kafka Sink Connector Post-Processors
+set to upsert mode.
+
+You can override the default write model by specifying a custom one in the
+``mongodb.writemodel.strategy`` configuration setting. The following
+strategies are provided with the connector:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Write Model
+     - Description
+
+   * - ReplaceOneDefaultStrategy
+     - | Replaces at most one document that matches the current document by the ``_id`` field.
+       | Default value for ``writemodel.strategy`` configuration setting.
+
+   * - ReplaceOneBusinessKeyStrategy
+     - | Replaces at most one document that matches filters provided by the ``document.id.strategy`` setting.
+       | Set the following configuration: ``writemodel.strategy=com.mongodb.kafka.connect.sink.writemodel.strategy.ReplaceOneBusinessKeyStrategy``
+
+       .. seealso:: Example of usage in :ref:`writemodel-strategy-business-key`.
+
+   * - DeleteOneDefaultStrategy
+     - | Deletes at most one document that contain null value sink records that matches the id specified by the ``document.id.strategy`` setting.
+       | Implicitly specified when the configuration setting ``mongodb.delete.on.null.values=true`` is set.
+       | You can set this explicitly with the following configuration: ``writemodel.strategy=com.mongodb.kafka.connect.sink.writemodel.strategy.DeleteOneDefaultStrategy``
+
+   * - UpdateOneTimestampsStrategy
+     - | Add ``_insertedTS`` (inserted timestamp) and ``_modifiedTS`` (modified timestamp) fields into documents.
+       | Set the following configuration: ``writemodel.strategy=com.mongodb.kafka.connect.sink.writemodel.strategy.UpdateOneTimestampsStrategy``
+
+       .. seealso:: Example of usage in :ref:`writemodel-strategy-timestamps`.
+
+.. note::
+
+   Future versions of the MongoDB Kafka Connector will allow developers to
+   specify a user-created, custom strategy in the configuration settings.
+
+.. _writemodel-strategy-business-key:
+
+WriteModel Strategy: Business Keys
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A business key is a value comprised of data within the sink record that
+identifies it as a unique document. This example shows how to define
+a business key using the data contained by multiple fields in the record as
+well as instruct the post processor to generate *BSON ObjectIds* for
+inserts, but not for updates.
+
+The following steps are required to configure this strategy:
+
+#. Create a **unique index** that corresponds to your business key in your
+   target MongoDB collection.
+#. In the connector configuration, specify the ``PartialValueStrategy``
+   as the id strategy to identify the fields that belong to the business
+   key.
+#. In the connector configuration, specify the
+   ``ReplaceOneBusinessKeyStrategy`` writemodel strategy.
+
+Suppose the parking usage data was being tracked for city blocks, defined
+by distinct values for ``street`` and ``avenue``. An example message
```

Comment:
Street and Avenue feel strange. Perhaps houseNumber and street? http://desktop.arcgis.com/en/arcmap/10.3/guide-books/geocoding/what-is-an-address.htm

Reply:
I am fine with any modification, but first would like to know whether you're requesting something different. I would like to adhere to the following goals regardless of the modification:

1) Connect the reader to an example they can conceptualize (rather than fieldA, fieldB which is in the currently published connector docs).
2) Keep the example as concise as possible. The minimum number of fields required to demonstrate a compound business key is two fields, and the minimum number of fields outside the business key is one.

That said, as mentioned in the example I am describing city blocks, so a simple substitution of houseNumber and street may not make sense for parking_spaces. Was there some other transient data that you were thinking of that I could use instead?

Reply:
I took this to mean "location".  The example immediately led me to think "what about parking spaces that aren't at an intersection".

Reply:
Ok, I will change the example.